source/agora/test/EnrollmentManager.d

```diff
@@ -167,3 +196,235 @@ unittest
         retryFor(node.hasPreimage(enroll.utxo_key,
             EnrollmentManager.PreimageRevealPeriod + 2), 5.seconds));
 }
+
+/// node which refuses to co-operate: doesn't sign / forges the signature / etc
+public class ByzantineNominator : Nominator
+{
+    /// SCP instance
+    private SCP* scp;
+
+    this (KeyPair key_pair, Ledger ledger,
+        TaskManager taskman, NetworkClient[PublicKey] peers,
+        SCPQuorumSet quorum_set)
+    {
+        super(key_pair, ledger, taskman, peers, quorum_set);
+    }
+
+    // refuse to nominate without enrollment
+    override void nominate (ulong slot_idx, ConsensusData prev,
+        ConsensusData next) @trusted
+    {
+        log.info("{}(): Proposing tx set for slot {}", __FUNCTION__, slot_idx);
+        if (next.enrolls.length > 0)
+            next.enrolls.length = 0;
+
+        auto prev_value = prev.serializeFull().toVec();
+        auto next_value = next.serializeFull().toVec();
+        if (this.scp.nominate(slot_idx, next_value, prev_value))
```

Comment:
Here is your segv. `this.scp` is never initialized.

Reply:
I changed it to `Protected SCP* scp;` on the Nominator.